packages/teams-js/test/public/app.spec.ts

```diff
@@ -11,585 +11,1213 @@ import {
+          expect(message.args.length).toBe(1);
+          expect(message.args[0]).toEqual(version);
+        });
+      });
+    });
+
+    describe('Testing app.notifySuccess function', () => {
+      it('app.notifySuccess should not allow calls before initialization', () => {
+        expect(() => app.notifySuccess()).toThrowError('The library has not yet been initialized');
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        it(`app.notifySuccess should successfully notify success with no error from ${context} context`, async () => {
+          await framelessPostMock.initializeWithContext(context);
+          app.notifySuccess();
+          const message = framelessPostMock.findMessageByFunc(app.Messages.Success);
+          expect(message).not.toBeNull();
+          expect(message.args.length).toBe(1);
+          expect(message.args[0]).toEqual(version);
+        });
+      });
+    });
+
+    describe('Testing app.notifyFailure function', () => {
+      it('app.notifyFailure should not allow calls before initialization', () => {
+        expect(() =>
+          app.notifyFailure({
+            reason: app.FailedReason.AuthFailed,
+            message: 'Failed message',
+          }),
+        ).toThrowError('The library has not yet been initialized');
+      });
+
+      Object.values(FrameContexts).forEach(context => {
+        it(`app.notifyFailure should call notify failure correctly with ${context} context`, async () => {
+          await framelessPostMock.initializeWithContext(context);
+
+          app.notifyFailure({
+            reason: app.FailedReason.AuthFailed,
+            message: 'Failed message',
+          });
+          const message = framelessPostMock.findMessageByFunc(app.Messages.Failure);
+          expect(message).not.toBeNull();
+          expect(message.args.length).toBe(2);
+          expect(message.args[0]).toEqual(app.FailedReason.AuthFailed);
+          expect(message.args[1]).toEqual('Failed message');
         });
+
+        it(`app.notifyFailure should call notify expected failure correctly with ${context} context`, async () => {
+          await framelessPostMock.initializeWithContext(context);
+
+          app.notifyExpectedFailure({
+            reason: app.ExpectedFailureReason.PermissionError,
+            message: 'Failed message',
+          });
+          const message = framelessPostMock.findMessageByFunc(app.Messages.ExpectedFailure);
+          expect(message).not.toBeNull();
+          expect(message.args.length).toBe(2);
+          expect(message.args[0]).toEqual(app.ExpectedFailureReason.PermissionError);
+          expect(message.args[1]).toEqual('Failed message');
+        });
+      });
+    });
+
+    describe('Testing app.registerOnThemeChangeHandler function', () => {
+      it('app.registerOnThemeChangeHandler should not allow calls before initialization', () => {
+        expect(() => app.registerOnThemeChangeHandler(() => {})).toThrowError(
```

Comment:
Can we define an emptyCallback and use it in the two places we're using this?

Reply:
And then we can add the `// eslint-disable-next-line @typescript-eslint/no-empty-function` above the emptyCallback declaration